### DISCUSSION TITLE
App size seems very big for a TODO app... could it be reduced?

### DISCUSSION THREAD

**Discussion by thekryz** [ransome1/sleek#658] [Ideas]
Sleek currently takes up 432 MBs as an app (on my Intel-Mac). That's 432 MBs to manage a sub-1MB-TODO-file. It's larger than Thunderbird (384 MB), Firefox (371 MB) and HandBrake (130 MB).

I feel like this is quite a number. And I would love to see this app become more light-weight. What do you think - would that be possible? Doesn't have to be top priority - but maybe something for the long run?

**Comment by ransome1** [comment]
> Sleek currently takes up 432 MBs as an app (on my Intel-Mac). That's 432 MBs to manage a sub-1MB-TODO-file. It's larger than Thunderbird (384 MB), Firefox (371 MB) and HandBrake (130 MB).

It is very hard to compare these apps (all native) with an Electron based app, which is literally being shipped with a headless version of Chromium. You'll find lot's of info on this on the web.

Plus, the Mac release of sleek is built an universal app, which means it contains releases for the Intel based AMD64 world as well as for the newer silicon based ARM64 one. 

There is a way of splitting the universal release into ARM and AMD releases, which will then be half the size. However the last time I tried this, the build process was failing in the GitHub pipeline. For the next build I can try to do the switch again, maybe it works now.

**Comment by thekryz** [comment]
I see your point. Would of course be nice to get rid of the Electron base and move to native, but I get that that's probably a very time-consuming task.

**Reply by ransome1** [reply]
I agree, this would be great to have sleek available as a native app. But it would be impossible for me to learn three new programming languages and maintain the code for three platforms. 

With the help of Electron I only need to speak one single language and press a button to build an app for all desktop platforms :) There are no plans of going to native route any time soon.

**Reply by ransome1** [reply]
v2.0.9 is ow available for both Apple platforms: https://github.com/ransome1/sleek/releases/tag/v2.0.9

**Reply by thekryz** [reply]
Thank you, that is a nice reduction! :)
